src/graph/sampler.cc

```diff
@@ -153,6 +155,7 @@ class EdgeSamplerObject: public Object {
 
   int64_t batch_size_;
   int64_t num_workers_;
+  bool replacement_;
```

Comment:
it seems you didn't change the sampling algorithm. if replacement=True, you should use sampling with replacement.